src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1763,160 +1766,177 @@ void HMICapabilitiesImpl::PrepareUiJsonValueForSaving(
+      const auto navigation_capability_so = navigation_capability();
+      if (navigation_capability_so) {
+        (*system_capabilities)[strings::navigation_capability] =
+            *navigation_capability_so;
+      }
     }
 
-    auto const video_streaming_capabilities = video_streaming_capability();
-    if (video_streaming_capabilities) {
-      (*system_capabilities)[strings::video_streaming_capability] =
-          *video_streaming_capabilities;
+    if (section_to_update == strings::phone_capability) {
+      const auto phone_capability_so = phone_capability();
+      if (phone_capability_so) {
+        (*system_capabilities)[strings::phone_capability] =
+            *phone_capability_so;
+      }
     }
 
-    save_hmi_capability_field_to_json(
-        strings::system_capabilities, schema, system_capabilities, out_node);
-  }
+    if (section_to_update == strings::video_streaming_capability) {
+      const auto video_streaming_capability_so = video_streaming_capability();
 
-  if (helpers::in_range(sections_to_update, strings::pcm_stream_capabilities)) {
-    save_hmi_capability_field_to_json(strings::pcm_stream_capabilities,
-                                      schema,
-                                      pcm_stream_capabilities(),
-                                      out_node);
-  }
+      if (video_streaming_capability_so) {
+        (*system_capabilities)[strings::video_streaming_capability] =
+            *video_streaming_capability_so;
+      }
+    }
+
+    if (section_to_update == strings::display_capabilities) {
+      const auto display_capabilities_so = display_capabilities();
+
+      if (display_capabilities_so) {
+        save_hmi_capability_field_to_json(strings::display_capabilities,
+                                          schema,
+                                          display_capabilities_so,
+                                          out_node);
+      }
+    }
+
+    if (section_to_update == strings::pcm_stream_capabilities) {
+      save_hmi_capability_field_to_json(strings::pcm_stream_capabilities,
+                                        schema,
+                                        pcm_stream_capabilities(),
+                                        out_node);
+    }
 
-  if (helpers::in_range(sections_to_update, hmi_response::language)) {
-    out_node[hmi_response::language] =
-        MessageHelper::CommonLanguageToString(active_ui_language());
+    if (section_to_update == hmi_response::language) {
+      out_node[hmi_response::language] =
+          MessageHelper::CommonLanguageToString(active_ui_language());
+    }
+
+    if (section_to_update == hmi_response::languages) {
+      save_hmi_capability_field_to_json(
+          hmi_response::languages, schema, ui_supported_languages(), out_node);
+    }
   }
 
-  if (helpers::in_range(sections_to_update, hmi_response::languages)) {
+  if (!system_capabilities->empty()) {
+    LOG4CXX_DEBUG(logger_, "KEK");
```

Comment:
@IGapchuk LOL

Reply:
@LitvinenkoIra Done in 7be5fccca8f77fc542cc64bcef5f6265e688b927